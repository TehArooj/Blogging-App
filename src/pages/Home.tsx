import { FormEvent, useEffect, useState } from "react";
import { signOutUser, auth, db } from "../utils/firebase/firebase.utils";
import { Link, useNavigate } from "react-router-dom";
import { HiSearch, HiOutlinePlusCircle } from "react-icons/hi";
import { FiLogOut } from "react-icons/fi";
import { FaBlog } from "react-icons/fa";
import { ImCancelCircle } from "react-icons/im";
import Modal from "react-modal";
import {
  collection,
  doc,
  DocumentData,
  getDocs,
  setDoc,
} from "firebase/firestore";
import uuid from "react-uuid";
import LoaderSpinner from "../components/LoaderSpinner.component";
import HomepageLoaderSpinner from "../components/HomepageLoaderSpinner.component";
Modal.setAppElement("#root");

const Home = () => {
  const navigate = useNavigate();
  const [username, setUsername] = useState("");
  const [uid, setUid] = useState("");
  const [modalIsOpen, setModalIsOpen] = useState(false);
  const [errorMsg, setErrorMsg] = useState("");
  const [successMsg, setSuccessMsg] = useState("");
  const [submitButtonDisabled, setSubmitButtonDisabled] = useState(false);
  const [gotData, setgotData] = useState(false);
  const [searchClicked, setSearchClicked] = useState(false);
  const [values, setValues] = useState({
    title: "",
    blog: "",
  });

  const [searchedData, setSearchedData] = useState<DocumentData[]>([]);
  const [blogData, setBlogData] = useState<DocumentData[]>([]);

  const [searchQuery, setSearchQuery] = useState("");
  const firstLetter = username.charAt(0).toUpperCase();

  const [isPhone, setIsPhone] = useState(
    window.matchMedia("(max-width: 767px)").matches
  );

  // Store username in local storage to maintain session
  useEffect(() => {
    window
      .matchMedia("(max-width: 767px)")
      .addEventListener("change", (e) => setIsPhone(e.matches));
  }, []);

  const [isDesktop, setIsDesktop] = useState(
    window.matchMedia("(min-width: 1024px)").matches
  );

  useEffect(() => {
    window
      .matchMedia("(min-width: 1024px)")
      .addEventListener("change", (e) => setIsDesktop(e.matches));
  }, []);

  // Store username in local storage to maintain session
  useEffect(() => {
    auth.onAuthStateChanged((user) => {
      if (user) {
        setUsername(user.displayName!);
        localStorage.setItem("userName", user.displayName!);
        setUid(user.uid);
      }
      if (!localStorage.getItem("userName")) {
        navigate("/login");
      }
      getData();
    });
  }, [navigate, username]);

  useEffect(() => {
    if (searchedData) {
      console.log(searchedData);
    }
  }, [searchedData]);

  // Signout user
  const SignOut = async () => {
    try {
      await signOutUser().then(() => {
        localStorage.clear();
        navigate("/login");
        console.log(auth);
      });
    } catch (error) {
      console.log(error);
    }
    console.log("Session end user logged out");
  };

  // Navigate to my blogs page
  const myBlogs = () => {
    navigate("/myblogs");
  };

  // Change input of title and blog
  const handleTitle = (e: FormEvent) => {
    e.preventDefault();
    setErrorMsg("");
    setValues((prev) => ({
      ...prev,
      title: (e.target as HTMLInputElement).value,
    }));
  };
  const handleBlog = (e: FormEvent) => {
    e.preventDefault();
    setErrorMsg("");
    setValues((prev) => ({
      ...prev,
      blog: (e.target as HTMLInputElement).value,
    }));
  };

  // Validation checks
  const validations = () => {
    if (!values.title || !values.blog) {
      setErrorMsg("Please fill all the fields.");
      setSubmitButtonDisabled(false);
      setSuccessMsg("");
      return false;
    }
    setErrorMsg("");
    return true;
  };

  // Submit a new Blog
  const handleSubmission = async (e: FormEvent) => {
    e.preventDefault();
    setSubmitButtonDisabled(true);
    // Validations
    const valid = validations();

    //Add new Blog
    if (valid) {
      let date = new Date().toUTCString();
      console.log(date);
      const title = values.title;
      const blog = values.blog;
      const id = uuid();
      await setDoc(doc(db, "blogs", id), {
        uid,
        username,
        date,
        title,
        blog,
        id,
      })
        .then((blogDocRef) => {
          setSubmitButtonDisabled(false);
          console.log("Document has been added successfully");
          setValues({
            blog: "",
            title: "",
          });
          setSuccessMsg("Document has been added successfully");
          setTimeout(() => {
            setSuccessMsg("");
          }, 3000);
          setModalIsOpen(false);
          getData();
        })
        .catch((err) => {
          setSubmitButtonDisabled(false);
          console.log(err.message);
        });
    }
  };

  // Get blogs data of all users
  const getData = async () => {
    const docsRef = collection(db, "blogs");
    const docsSnap = await getDocs(docsRef);
    if (docsSnap) {
      let newBlogsData: DocumentData[] = [];
      docsSnap.docs.forEach((doc) => {
        newBlogsData.push(doc.data());
      });
      console.log(newBlogsData);
      newBlogsData.sort((a, b) =>
        new Date(a.date) < new Date(b.date) ? 1 : -1
      );
      setBlogData(newBlogsData);
      setgotData(true);
    } else {
      setgotData(false);
      console.log("No documents found.");
    }
    setSearchQuery("");
  };

  // Search
  const handleSearch = (e: FormEvent) => {
    e.preventDefault();
    setSearchClicked(!searchClicked);
    console.log("Search");
  };

  // Close Modal and refresh to get realtime data
  const closeModalAndRefreshPage = () => {
    getData();
    setModalIsOpen(false);
  };
  useEffect(() => {
    console.log(searchQuery);
    console.log(searchedData);
    setSearchedData(
      blogData.filter((blog) =>
        blog.title.toLowerCase().includes(searchQuery.toLowerCase())
      )
    );
  }, [blogData, searchQuery]);

  // Format date
  const formatDate = (d: Date, format: boolean) => {
    let ye = new Intl.DateTimeFormat("en", { year: "numeric" }).format(d);
    let mo = new Intl.DateTimeFormat("en", { month: "long" }).format(d);
    let da = new Intl.DateTimeFormat("en", { day: "2-digit" }).format(d);

    return format ? `${da} ${mo} ${ye}` : `${da} ${mo}`;
  };

  return (
    <>
      <div className="fixed top-0 left-0 flex justify-around items-center w-24 h-screen bg-darkGrey  flex-col md:flex-row md:bottom-0 md:h-20 md:w-screen md:top-auto md:drop-shadow-[0_-6mm_4mm_white]  tb:flex-row tb:bottom-0 tb:h-16 tb:w-screen tb:top-auto  tb:drop-shadow-[0_-6mm_4mm_white]   m:flex-row m:bottom-0 m:h-16 m:w-screen m:top-auto m:drop-shadow-[0_-6mm_4mm_white]">
        <div className="flex flex-col justify-between h-full w-full mt-4 mb-4 items-center md:flex-row tb:flex-row m:flex-row">
          <div className="md:flex md:flex-row md:w-full md:justify-evenly tb:justify-evenly tb:flex tb:flex-row tb:w-full m:justify-evenly m:flex m:flex-row m:w-full">
            <div className=" md:flex tb:flex m:flex items-center text-center font-semibold ">
              <span className="ml-2 w-16 h-16 block text-2xl text-white text-center rounded-50  bg-primary p-4 md:w-9 md:h-9 md:p-1 md:text-xl  tb:w-9 tb:h-9 tb:p-1 tb:text-xl m:w-9 m:h-9 m:p-1 m:text-xl">
                {firstLetter}
              </span>
              <div className="2xl:hidden xl:hidden lg:hidden md:visible md:text-sm md:ml-3 md:font-normal md:text-white tb:hidden  m:hidden">
                {username}
              </div>
            </div>

            <div
              className=" items-center 2xl:mt-5 xl:mt-5 lg:mt-5  md:flex md:justify-between cursor-pointer hover:bg-darkerGrey p-4 rounded-lg transition-all duration-200"
              onClick={handleSearch}
            >
              <HiSearch className="ml-2 text-center text-3xl text-primary tb:text-2xl tb:ml-0 m:ml-0 m:text-2xl" />
              <span className=" text-sm text-white ml-1 tb:hidden m:hidden ">
                search
              </span>
            </div>

            <div
              className=" items-center 2xl:mt-5 xl:mt-5 lg:mt-5 md:flex md:justify-between cursor-pointer hover:bg-darkerGrey p-4 rounded-lg transition-all duration-200 "
              onClick={() => setModalIsOpen(true)}
            >
              <HiOutlinePlusCircle className="ml-2  text-3xl text-primary  tb:text-2xl tb:ml-0 m:ml-0 m:text-2xl" />
              <span className=" text-sm text-white ml-1 tb:hidden m:hidden">
                create
              </span>
            </div>
            <Modal
              isOpen={modalIsOpen}
              shouldCloseOnOverlayClick={false}
              onRequestClose={() => setModalIsOpen(false)}
              style={{
                overlay: {
                  right: "0",
                  left: "0",
                  backgroundColor: "rgb(0,0,0,0.5)",
                },
                content: {
                  padding: "0px",
                  borderColor: "rgb(0,0,0,.5)",
                  borderWidth: "1px",
                  borderRadius: "16px",
                  marginLeft: "auto",
                  marginRight: "auto",
                  marginTop: isPhone ? "3px" : "20px",
                  width: isPhone ? "80%" : "58%",
                  height: isPhone ? "85%" : "80%",
                },
              }}
            >
              <>
                {submitButtonDisabled ? (
                  <LoaderSpinner />
                ) : (
                  <>
                    <div className="flex justify-end tb:mt-4 m:mt-4 2xl:hidden xl:hidden lg:hidden md:hidden">
                      <div className="absolute tb:mr-5 tb:p-1 m:mr-3 m:p-0">
                        <div
                          onClick={closeModalAndRefreshPage}
                          className="cursor-pointer"
                        >
                          <ImCancelCircle className="text-2xl text-secondary  hover:text-primary  tb:text-base m:text-base " />
                        </div>
                      </div>
                    </div>
                    <div className="grid grid-cols-12 ">
                      <div className="col-span-12 mt-8 pl-8 pr-8 m:pl-4 m:pr-4 tb:pl-5 tb:pr-5 tb:items-center tb:justify-center m:items-center m:justify-center">
                        <div className="flex justify-between tb:justify-center m:justify-center">
                          <h1 className="mb-2 font-dm font-bold text-4xl text-left  text-darkGrey tb:text-center m:text-center">
                            New Blog
                          </h1>
                          <div className="mt-1 mr-5 tb:hidden m:hidden">
                            <div className="absolute">
                              <div
                                onClick={closeModalAndRefreshPage}
                                className="cursor-pointer"
                              >
                                <ImCancelCircle className="text-2xl text-secondary  hover:text-primary  tb:text-base m:text-base " />
                              </div>
                            </div>
                          </div>
                        </div>
                        <div className="font-lexend flex flex-col ">
                          <form>
                            <div className="text-secondary ">
                              <p className="text-xl font-light mb-6 tb:text-center tb:text-base m:text-center m:text-base ">
                                Let's show the world what you have for them
                              </p>
                              <input
                                className="text-darkGrey border-solid border-2 border-secondary pt-4 pb-4 pl-8 pr-8 mb-2 w-full focus:outline-none focus:border-primary  tb:pl-4 tb:text-sm m:text-xs m:pl-4 "
                                type="text"
                                required
                                placeholder="Title"
                                value={values.title}
                                onChange={handleTitle}
                              />
                              <textarea
                                className="text-darkGrey border-solid border-2 border-secondary p-5 mb-2 w-full  focus:outline-none focus:border-primary tb:pl-4 tb:text-sm  m:text-xs m:pl-4"
                                placeholder="Write Blog Details"
                                required
                                value={values.blog}
                                onChange={handleBlog}
                                rows={12}
                                cols={12}
                              />
                              <b className=" text-sm text-errorMsg mb-5  ">
                                {errorMsg}
                              </b>
                              <b className=" text-sm text-successMsg mb-5  ">
                                {successMsg}
                              </b>
                              <div className="flex justify-end tb:justify-center m:justify-center">
                                <button
                                  className="text-white font-semibold bg-secondary border-solid border-2  border-secondary  h-14 w-44  hover:outline-none hover:bg-darkGrey hover:border-none  disabled:bg-gray-500  tb:h-10  m:w-full m:h-10 m:text-sm"
                                  type="submit"
                                  onClick={handleSubmission}
                                  disabled={submitButtonDisabled}
                                >
                                  SUBMIT
                                </button>
                              </div>
                            </div>
                          </form>
                        </div>
                      </div>
                    </div>
                  </>
                )}
              </>
            </Modal>
            <div
              className="2xl:mt-5 xl:mt-5 lg:mt-5 items-center md:flex md:justify-between cursor-pointer hover:bg-darkerGrey p-4 rounded-lg transition-all duration-200"
              onClick={myBlogs}
            >
              <FaBlog className="ml-3 text-2xl text-primary tb:text-xl tb:ml-1 m:ml-1 m:text-xl" />
              <span className=" text-sm text-white ml-1 tb:hidden m:hidden">
                blogs
              </span>
            </div>
          </div>
          <div
            className="items-center md:w-30 md:mt-0 md:mr-10 md:flex md:justify-between cursor-pointer hover:bg-darkerGrey p-4 rounded-lg transition-all duration-200	 "
            onClick={SignOut}
          >
            <FiLogOut className="ml-2 text-3xl text-primary md:text-5xl tb:text-2xl tb:ml-0 tb:mr-14 m:mr-6 m:ml-0 m:text-2xl" />
            <span className=" text-sm text-white ml-1 tb:hidden m:hidden w-full">
              logout
            </span>
          </div>
        </div>
      </div>

      <div
        className={`absolute top-0 left-0 w-screen h-20 m:h-16 ml-24 md:ml-0 tb:ml-0 m:ml-0 flex items-center flex-row justify-center bg-darkGrey  drop-shadow-2xl ${
          searchClicked ? "block" : "hidden"
        }`}
      >
        <input
          value={searchQuery}
          placeholder="Search By Title"
          type="search"
          className=" w-1/2 p-3 rounded-sm outline-none m:p-2"
          onChange={(e) => setSearchQuery(e.target.value)}
        />
        <div className="ml-8 m:ml-4 tb:ml-4 m:p-3 cursor-pointer hover:bg-darkerGrey p-4 rounded-lg transition-all duration-200 ">
          <ImCancelCircle
            className=" text-primary text-xl"
            onClick={handleSearch}
          />
        </div>
      </div>
      <div className=" flex flex-col mt-10 md:mb-20 tb:mb-16 m:mb-14 ml-24 pl-28 pr-28  md:ml-0 tb:ml-0 m:ml-0 md:pl-14 md:pr-14 tb:pl-12 tb:pr-12 m:mt-20 m:pl-11 m:pr-11">
        <div
          className={`mb-10 flex flex-col items-left ${
            searchClicked ? "mt-20" : ""
          }`}
        >
          <div className="m:flex m:justify-center m:items-center ">
            <div className="bg-primary pt-1 pb-1 w-5"></div>
          </div>
          <div className="text-xl font-lexend font-normal m:text-center ">
            Latest
          </div>

          <div className="flex flex-col mt-10 items-left ">
            {gotData ? (
              <>
                {searchedData.length > 0 &&
                  searchedData.map((item) => {
                    return (
                      <div key={item.id}>
                        <h1 className="text-2xl font-semibold mb-1 m:hidden">
                          {formatDate(new Date(item.date), false).toUpperCase()}
                        </h1>
                        <Link to={`/viewblog/${item.id}`} className="mb-14">
                          <h1 className="text-4xl text-primary text-left font-dm font-normal mb-5 m:mb-3 tb:text-3xl  m:text-2xl">
                            {item.title}
                          </h1>
                          <p className="font-normal text-left m:text-base m:font-extralight ">
                            {item.blog.slice(0, 650)}
                            {item.blog.length > 650 && (
                              <span>
                                ...{" "}
                                <span className="text-primary font-normal">
                                  read more
                                </span>
                              </span>
                            )}
                          </p>
                        </Link>
                        <div className="flex justify-between mb-8 mt-4 tb:mb-12 m:mb-12   ">
                          <div className=" 2xl:hidden xl:hidden lg:hidden md:hidden tb:hidden m:visible m:text-base m:font-semibold ">
                            {formatDate(
                              new Date(item.date),
                              true
                            ).toUpperCase()}
                          </div>
                          <div className="text-secondary text-base font-light m:text-right ">
                            @{item.username}
                          </div>
                        </div>
                      </div>
                    );
                  })}{" "}
              </>
            ) : isDesktop ? (
              <HomepageLoaderSpinner />
            ) : (
              <LoaderSpinner />
            )}
          </div>
        </div>
      </div>
    </>
  );
};

export default Home;
