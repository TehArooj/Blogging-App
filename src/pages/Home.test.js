import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { cleanup } from "@testing-library/react";
import Home from "./Home";
import LoaderSpinner from "../components/LoaderSpinner.component"
import HomepageLoaderSpinner from "../components/HomepageLoaderSpinner.component";

afterEach(cleanup);
window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      addListener: () => {},
      removeListener: () => {},
      addEventListener: () => {},
    };
  };

it("<Home /> renders correctly", async () => {
  const tree = renderer
    .create(
      <Router>
        <Home />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("<Home /> renders correctly when there is a Loader Spinner", () => {
  const tree = renderer.create(<LoaderSpinner />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("<Home /> renders correctly when there is a  Homepage Loader Spinner", () => {
  const tree = renderer.create(<HomepageLoaderSpinner />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("<Home /> check matchers and pass", async () => {
  const blog = {
    date: new Date(),
    blog: "Blog Text",
    title: "LeBron James new book stars",
    author: "Maxwell",
  };

  expect(blog).toMatchSnapshot({
    date: expect.any(Date),
    blog: expect.any(String),
    title: expect.any(String),
    author: expect.any(String),
  });
});
