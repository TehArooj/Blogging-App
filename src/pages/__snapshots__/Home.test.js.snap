// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> check matchers and pass 1`] = `
Object {
  "author": Any<String>,
  "blog": Any<String>,
  "date": Any<Date>,
  "title": Any<String>,
}
`;

exports[`<Home /> renders correctly 1`] = `
Array [
  <div
    className="fixed top-0 left-0 flex justify-around items-center w-24 h-screen bg-darkGrey  flex-col md:flex-row md:bottom-0 md:h-20 md:w-screen md:top-auto md:drop-shadow-[0_-6mm_4mm_white]  tb:flex-row tb:bottom-0 tb:h-16 tb:w-screen tb:top-auto  tb:drop-shadow-[0_-6mm_4mm_white]   m:flex-row m:bottom-0 m:h-16 m:w-screen m:top-auto m:drop-shadow-[0_-6mm_4mm_white]"
  >
    <div
      className="flex flex-col justify-between h-full w-full mt-4 mb-4 items-center md:flex-row tb:flex-row m:flex-row"
    >
      <div
        className="md:flex md:flex-row md:w-full md:justify-evenly tb:justify-evenly tb:flex tb:flex-row tb:w-full m:justify-evenly m:flex m:flex-row m:w-full"
      >
        <div
          className=" md:flex tb:flex m:flex items-center text-center font-semibold "
        >
          <span
            className="ml-2 w-16 h-16 block text-2xl text-white text-center rounded-50  bg-primary p-4 md:w-9 md:h-9 md:p-1 md:text-xl  tb:w-9 tb:h-9 tb:p-1 tb:text-xl m:w-9 m:h-9 m:p-1 m:text-xl"
          />
          <div
            className="2xl:hidden xl:hidden lg:hidden md:visible md:text-sm md:ml-3 md:font-normal md:text-white tb:hidden  m:hidden"
          />
        </div>
        <div
          className=" items-center 2xl:mt-5 xl:mt-5 lg:mt-5  md:flex md:justify-between cursor-pointer hover:bg-darkerGrey p-4 rounded-lg transition-all duration-200"
          onClick={[Function]}
        >
          <svg
            className="ml-2 text-center text-3xl text-primary tb:text-2xl tb:ml-0 m:ml-0 m:text-2xl"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
              fillRule="evenodd"
            />
          </svg>
          <span
            className=" text-sm text-white ml-1 tb:hidden m:hidden "
          >
            search
          </span>
        </div>
        <div
          className=" items-center 2xl:mt-5 xl:mt-5 lg:mt-5 md:flex md:justify-between cursor-pointer hover:bg-darkerGrey p-4 rounded-lg transition-all duration-200 "
          onClick={[Function]}
        >
          <svg
            className="ml-2  text-3xl text-primary  tb:text-2xl tb:ml-0 m:ml-0 m:text-2xl"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 9v3m0 0v3m0-3h3m-3 0H9m12 0a9 9 0 11-18 0 9 9 0 0118 0z"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
          <span
            className=" text-sm text-white ml-1 tb:hidden m:hidden"
          >
            create
          </span>
        </div>
        <div
          className="2xl:mt-5 xl:mt-5 lg:mt-5 items-center md:flex md:justify-between cursor-pointer hover:bg-darkerGrey p-4 rounded-lg transition-all duration-200"
          onClick={[Function]}
        >
          <svg
            className="ml-3 text-2xl text-primary tb:text-xl tb:ml-1 m:ml-1 m:text-xl"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M172.2 226.8c-14.6-2.9-28.2 8.9-28.2 23.8V301c0 10.2 7.1 18.4 16.7 22 18.2 6.8 31.3 24.4 31.3 45 0 26.5-21.5 48-48 48s-48-21.5-48-48V120c0-13.3-10.7-24-24-24H24c-13.3 0-24 10.7-24 24v248c0 89.5 82.1 160.2 175 140.7 54.4-11.4 98.3-55.4 109.7-109.7 17.4-82.9-37-157.2-112.5-172.2zM209 0c-9.2-.5-17 6.8-17 16v31.6c0 8.5 6.6 15.5 15 15.9 129.4 7 233.4 112 240.9 241.5.5 8.4 7.5 15 15.9 15h32.1c9.2 0 16.5-7.8 16-17C503.4 139.8 372.2 8.6 209 0zm.3 96c-9.3-.7-17.3 6.7-17.3 16.1v32.1c0 8.4 6.5 15.3 14.8 15.9 76.8 6.3 138 68.2 144.9 145.2.8 8.3 7.6 14.7 15.9 14.7h32.2c9.3 0 16.8-8 16.1-17.3-8.4-110.1-96.5-198.2-206.6-206.7z"
            />
          </svg>
          <span
            className=" text-sm text-white ml-1 tb:hidden m:hidden"
          >
            blogs
          </span>
        </div>
      </div>
      <div
        className="items-center md:w-30 md:mt-0 md:mr-10 md:flex md:justify-between cursor-pointer hover:bg-darkerGrey p-4 rounded-lg transition-all duration-200	 "
        onClick={[Function]}
      >
        <svg
          className="ml-2 text-3xl text-primary md:text-5xl tb:text-2xl tb:ml-0 tb:mr-14 m:mr-6 m:ml-0 m:text-2xl"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
          />
          <polyline
            points="16 17 21 12 16 7"
          />
          <line
            x1="21"
            x2="9"
            y1="12"
            y2="12"
          />
        </svg>
        <span
          className=" text-sm text-white ml-1 tb:hidden m:hidden w-full"
        >
          logout
        </span>
      </div>
    </div>
  </div>,
  <div
    className="absolute top-0 left-0 w-screen h-20 m:h-16 ml-24 md:ml-0 tb:ml-0 m:ml-0 flex items-center flex-row justify-center bg-darkGrey  drop-shadow-2xl hidden"
  >
    <input
      className=" w-1/2 p-3 rounded-sm outline-none m:p-2"
      onChange={[Function]}
      placeholder="Search By Title"
      type="search"
      value=""
    />
    <div
      className="ml-8 m:ml-4 tb:ml-4 m:p-3 cursor-pointer hover:bg-darkerGrey p-4 rounded-lg transition-all duration-200 "
    >
      <svg
        className=" text-primary text-xl"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        version="1.1"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
        />
        <path
          d="M10.5 4l-2.5 2.5-2.5-2.5-1.5 1.5 2.5 2.5-2.5 2.5 1.5 1.5 2.5-2.5 2.5 2.5 1.5-1.5-2.5-2.5 2.5-2.5z"
        />
      </svg>
    </div>
  </div>,
  <div
    className=" flex flex-col mt-10 md:mb-20 tb:mb-16 m:mb-14 ml-24 pl-28 pr-28  md:ml-0 tb:ml-0 m:ml-0 md:pl-14 md:pr-14 tb:pl-12 tb:pr-12 m:mt-20 m:pl-11 m:pr-11"
  >
    <div
      className="mb-10 flex flex-col items-left "
    >
      <div
        className="m:flex m:justify-center m:items-center "
      >
        <div
          className="bg-primary pt-1 pb-1 w-5"
        />
      </div>
      <div
        className="text-xl font-lexend font-normal m:text-center "
      >
        Latest
      </div>
      <div
        className="flex flex-col mt-10 items-left "
      >
        <div
          className="absolute top-[45%] left-[45%] m:top-[40%] m:left-[40%]"
        >
          <div
            aria-busy={true}
            aria-label="ball-triangle-loading"
            data-testid="ball-triangle-loading"
            role="status"
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <svg
              data-testid="ball-triangle-svg"
              height={100}
              stroke="#4fa94d"
              viewBox="0 0 57 57"
              width={100}
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fillRule="evenodd"
              >
                <g
                  strokeWidth="2"
                  transform="translate(1 1)"
                >
                  <circle
                    cx="5"
                    cy="50"
                    r={5}
                  >
                    <animate
                      attributeName="cy"
                      begin="0s"
                      calcMode="linear"
                      dur="2.2s"
                      repeatCount="indefinite"
                      values="50;5;50;50"
                    />
                    <animate
                      attributeName="cx"
                      begin="0s"
                      calcMode="linear"
                      dur="2.2s"
                      repeatCount="indefinite"
                      values="5;27;49;5"
                    />
                  </circle>
                  <circle
                    cx="27"
                    cy="5"
                    r={5}
                  >
                    <animate
                      attributeName="cy"
                      begin="0s"
                      calcMode="linear"
                      dur="2.2s"
                      from="5"
                      repeatCount="indefinite"
                      to="5"
                      values="5;50;50;5"
                    />
                    <animate
                      attributeName="cx"
                      begin="0s"
                      calcMode="linear"
                      dur="2.2s"
                      from="27"
                      repeatCount="indefinite"
                      to="27"
                      values="27;49;5;27"
                    />
                  </circle>
                  <circle
                    cx="49"
                    cy="50"
                    r={5}
                  >
                    <animate
                      attributeName="cy"
                      begin="0s"
                      calcMode="linear"
                      dur="2.2s"
                      repeatCount="indefinite"
                      values="50;50;5;50"
                    />
                    <animate
                      attributeName="cx"
                      begin="0s"
                      calcMode="linear"
                      dur="2.2s"
                      from="49"
                      repeatCount="indefinite"
                      to="49"
                      values="49;5;27;49"
                    />
                  </circle>
                </g>
              </g>
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`<Home /> renders correctly when there is a  Homepage Loader Spinner 1`] = `
<div
  className="absolute top-[45%] left-1/2"
>
  <div
    aria-busy={true}
    aria-label="ball-triangle-loading"
    data-testid="ball-triangle-loading"
    role="status"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <svg
      data-testid="ball-triangle-svg"
      height={100}
      stroke="#4fa94d"
      viewBox="0 0 57 57"
      width={100}
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <g
          strokeWidth="2"
          transform="translate(1 1)"
        >
          <circle
            cx="5"
            cy="50"
            r={5}
          >
            <animate
              attributeName="cy"
              begin="0s"
              calcMode="linear"
              dur="2.2s"
              repeatCount="indefinite"
              values="50;5;50;50"
            />
            <animate
              attributeName="cx"
              begin="0s"
              calcMode="linear"
              dur="2.2s"
              repeatCount="indefinite"
              values="5;27;49;5"
            />
          </circle>
          <circle
            cx="27"
            cy="5"
            r={5}
          >
            <animate
              attributeName="cy"
              begin="0s"
              calcMode="linear"
              dur="2.2s"
              from="5"
              repeatCount="indefinite"
              to="5"
              values="5;50;50;5"
            />
            <animate
              attributeName="cx"
              begin="0s"
              calcMode="linear"
              dur="2.2s"
              from="27"
              repeatCount="indefinite"
              to="27"
              values="27;49;5;27"
            />
          </circle>
          <circle
            cx="49"
            cy="50"
            r={5}
          >
            <animate
              attributeName="cy"
              begin="0s"
              calcMode="linear"
              dur="2.2s"
              repeatCount="indefinite"
              values="50;50;5;50"
            />
            <animate
              attributeName="cx"
              begin="0s"
              calcMode="linear"
              dur="2.2s"
              from="49"
              repeatCount="indefinite"
              to="49"
              values="49;5;27;49"
            />
          </circle>
        </g>
      </g>
    </svg>
  </div>
</div>
`;

exports[`<Home /> renders correctly when there is a Loader Spinner 1`] = `
<div
  className="absolute top-[45%] left-[45%] m:top-[40%] m:left-[40%]"
>
  <div
    aria-busy={true}
    aria-label="ball-triangle-loading"
    data-testid="ball-triangle-loading"
    role="status"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <svg
      data-testid="ball-triangle-svg"
      height={100}
      stroke="#4fa94d"
      viewBox="0 0 57 57"
      width={100}
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <g
          strokeWidth="2"
          transform="translate(1 1)"
        >
          <circle
            cx="5"
            cy="50"
            r={5}
          >
            <animate
              attributeName="cy"
              begin="0s"
              calcMode="linear"
              dur="2.2s"
              repeatCount="indefinite"
              values="50;5;50;50"
            />
            <animate
              attributeName="cx"
              begin="0s"
              calcMode="linear"
              dur="2.2s"
              repeatCount="indefinite"
              values="5;27;49;5"
            />
          </circle>
          <circle
            cx="27"
            cy="5"
            r={5}
          >
            <animate
              attributeName="cy"
              begin="0s"
              calcMode="linear"
              dur="2.2s"
              from="5"
              repeatCount="indefinite"
              to="5"
              values="5;50;50;5"
            />
            <animate
              attributeName="cx"
              begin="0s"
              calcMode="linear"
              dur="2.2s"
              from="27"
              repeatCount="indefinite"
              to="27"
              values="27;49;5;27"
            />
          </circle>
          <circle
            cx="49"
            cy="50"
            r={5}
          >
            <animate
              attributeName="cy"
              begin="0s"
              calcMode="linear"
              dur="2.2s"
              repeatCount="indefinite"
              values="50;50;5;50"
            />
            <animate
              attributeName="cx"
              begin="0s"
              calcMode="linear"
              dur="2.2s"
              from="49"
              repeatCount="indefinite"
              to="49"
              values="49;5;27;49"
            />
          </circle>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
